Adding support for MariaDB.

renemadsen opened the pull request:
Adding support for MariaDBs MARIADB_ROOT_PASSWORD.

HofmeisterAn commented on src/Testcontainers/Configurations/Modules/Databases/MariaDbTestcontainerConfiguration.cs:
```diff
@@ -0,0 +1,66 @@
+    private const int MariaDbPort = 3306;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MySqlTestcontainerConfiguration" /> class.
```
`<see cref="MariaDbTestcontainerConfiguration" />`

HofmeisterAn commented on src/Testcontainers/Configurations/Modules/Databases/MariaDbTestcontainerConfiguration.cs:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MySqlTestcontainerConfiguration" /> class.
```
`<see cref="MariaDbTestcontainerConfiguration" />`

HofmeisterAn commented on src/Testcontainers/Containers/Modules/Databases/MariaDbTestcontainer.cs:
```diff
@@ -0,0 +1,45 @@
+  public sealed class MariaDbTestcontainer : TestcontainerDatabase
+  {
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MySqlTestcontainer" /> class.
```
`<see cref="MariaDbTestcontainer" />`

HofmeisterAn reviewed (commented):
LGTM, but I know Sonar will complain. Can't we inherit from the MySql classes (`MySqlTestcontainerConfiguration`, `MariaDbTestcontainer`)?

I can take care of the XML comments too.

renemadsen commented:
> LGTM, but I know Sonar will complain.

Why will Sonar complain?

Fixed the summary comments.
Also switched to using Pomelo.EntityFrameworkCore.MySql, since it's being maintained openly on github whereas the MySql.Data is maintaned by Oracle. This affects the MySQL testing container also.

HofmeisterAn commented:
> Why will Sonar complain?

Code duplication.

HofmeisterAn reviewed (approved):
👏 I'll merge it tomorrow — thanks. If you like, you can squash the commits and use a semantic commit message to follow the repository schema.

_I just notice that we skip those paths regarding code duplication._

renemadsen committed:
feat: Adding support for MariaDB.
feat: Adding support for MariaDBs MARIADB_ROOT_PASSWORD.
feat: Changing MySQL and MariaDB to use Pomelo.EntityFrameworkCore.MySql, since it's more up to date for both DB types.

HofmeisterAn commented:
https://github.com/testcontainers/testcontainers-dotnet/commit/f5d471cd93797abb9a1a8108a9eae17db6d063da.
